src/main/java/org/interledger/cryptoconditions/FulfillmentBase.java

```diff
@@ -0,0 +1,66 @@
+package org.interledger.cryptoconditions;
+
+import java.util.EnumSet;
+
+
+import org.interledger.cryptoconditions.encoding.Base64Url;
+
+public abstract class FulfillmentBase  implements Fulfillment {
+
+	protected final byte[] payload;	
+	protected final Condition condition;
+
+	/*
+	 * Default constructor. Raise exception to force use of child classes.
+	 */
+	FulfillmentBase() {
```

Comment:
This is one of the things I don't really like and would like to refactor (using a Collaborator pattern with no abstract and child classes). The idea was to force a common behavior that makes payload and condition immutable in all types of Fulfillments.